File: 1877-minimize-maximum-pair-sum-in-array/1877-minimize-maximum-pair-sum-in-array.java
class Solution {
    public int minPairSum(int[] nums) {
        int ans = 0;
		Arrays.sort(nums);
		int l = 0, r = nums.length - 1, indx = 0;
		
		int arr[] = new int [nums.length / 2];
		
		while (r > l)
		{
			arr[indx] = (nums[l] + nums[r]);
			indx++; l++; r--;
		}
		
		ans = Arrays.stream(arr).max().getAsInt();
		
		return ans;
    }
}

File: 1877-minimize-maximum-pair-sum-in-array/1877-minimize-maximum-pair-sum-in-array.py
class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        nums.sort()
        max_pair = 0
        for i in range(1, len(nums) + 1):
            max_pair = max(max_pair, nums[i - 1] + nums[-i])
        return max_pair